azavea/echo-locator
# Switch frontend app bundler

## Overview

This PR has made the following changes as a result of switching away from `mastarm`:
- [X] move `taui` and rename to `src/frontend`
- [X] remove `mastarm` and its related config from `./taui/configurations`
- [X] switch to webpack + babel and add configs for them
- [X] update paths and volume mounting in `docker-compose` for `app` and `django` services
- [X] serve built frontend as static assets from django root
- [X] hard-code redux store init values from what used to be in `store.yml`
- [X] read from `.env` on frontend
- [X] update task runner scripts in `package.json`
- [X] update imports in JS due to changed directory structure
- [x] linter and formater
- [x] update CI/CD script if need be

### Demo

App served from Django root
![Screen Shot 2022-04-26 at 6 18 40 PM](https://user-images.githubusercontent.com/16109558/165402022-d7b1eb8e-be15-4a55-8388-6820194acf89.png)

App in the map view
![Screen Shot 2022-04-26 at 6 18 56 PM](https://user-images.githubusercontent.com/16109558/165402055-b4c88fde-4abb-4d24-9b2d-6f7f864e2419.png)

### Notes
1. I tried to keep the formatter change and the linter change separate from the meaningful code changes
2. The following will be in their own issues:
- handle env-based dotenv files better, pull dotenv file in bootstrap script
- favicons

## Testing Instructions

 * Pull down this branch
 * At the root of the repo, run: `cp ./src/frontend/.env.template .env`. Then fill in the fields in `.env` by the one stored on lastpass
 * At the root of the repo, run `./scripts/update`, which will build the frontend and backend, install frontend deps, create a static build of the frontend to be used by django, and run django manage scripts
 * AT the root of the repo, run `./scripts/server`, which will bring both frontend and backend up and running. Navigate to `http://localhost:9966` to access app frontend. If you make changes to frontend code, module hot reloading should work. Navigate to `http://localhost:8085` to access the static frontend app served from django.


Resolves #376
